pkg/controller/kieapp/kieapp_controller.go

```diff
@@ -185,30 +193,72 @@ func (reconciler *ReconcileKieApp) Reconcile(request reconcile.Request) (reconci
 	if len(dcUpdates) > 0 {
 		for _, uDc := range dcUpdates {
 			newDC := uDc.DeepCopyObject()
-			rResult, err := reconciler.updateObj(uDc.Name, uDc.Namespace, newDC)
+			logrus.Infof("Updating %s %s/%s\n", uDc.Kind, uDc.Namespace, uDc.Name)
+			rResult, err := reconciler.updateObj(newDC)
 			if err != nil {
 				return rResult, err
 			}
 		}
-		return reconcile.Result{Requeue: true}, nil
+		return rResult, nil
 	}
 
 	// Update status.Deployments if needed
 	if !reflect.DeepEqual(dcNames, instance.Status.Deployments) {
 		instance.Status.Deployments = dcNames
-		return reconciler.crUpdate(instance)
+		return reconciler.updateObj(instance)
 	}
 
 	return rResult, nil
 }
 
-func (reconciler *ReconcileKieApp) crUpdate(cr *v1.KieApp) (reconcile.Result, error) {
-	err := reconciler.client.Update(context.TODO(), cr)
+// Check ImageStream
+func (reconciler *ReconcileKieApp) checkImageStreamTag(name, namespace string) bool {
+	result := strings.Split(name, ":")
+	if len(result) == 1 {
+		result = append(result, "latest")
+	}
+	tagName := fmt.Sprintf("%s:%s", result[0], result[1])
+	_, err := reconciler.imageClient.ImageStreamTags(namespace).Get(tagName, metav1.GetOptions{})
 	if err != nil {
-		logrus.Printf("failed to update instance status: %v", err)
-		return reconcile.Result{}, err
+		return false
 	}
-	return reconcile.Result{Requeue: true}, nil
+	return true
+}
+
+// Create ImageStreamTag
+func (reconciler *ReconcileKieApp) createLocalImageTag(currentTagReference corev1.ObjectReference, cr *v1.KieApp) error {
+	if !reconciler.checkImageStreamTag(currentTagReference.Name, cr.Namespace) {
```

Comment:
Isn't this a duplicate check? Seems you called this function because IS did not exist. Also, the function should be meaningful on its own, and its job is to create the local image tag, not to create it "if not exists"... right?